src/InvoiceGenerator/InvoiceGenerator/ClientDetailsEntryScreen.cs

```diff
@@ -19,31 +20,170 @@ public ClientDetailsEntryScreen()
         #region ButtonClicks
+        private void btn_enterDetails_Click(object sender, EventArgs e)
+        {
+            pnl_viewDetails.Hide();
+            pnl_enterDetails.Show();
+        }
+
+        private void btn_viewDetails_Click(object sender, EventArgs e)
+        {
+            pnl_enterDetails.Hide();
+            pnl_viewDetails.Show();
+        }
+
         private void btn_Create_Click(object sender, EventArgs e)
         {
             using (var context = new InvoiceDBContext())    // using statement is discarded once the squiggly brackets have been passed, saves memory
             {
                 context.Database.EnsureCreated();   // Ensures database is created
 
-                var client = new Client     // Acts as an intialiser
+                bool validUserDetails = validateUserDetails();
+                bool validEmailFormat = validateEmailFormatting();
+
+                if (validUserDetails && validEmailFormat)
                 {
-                    ClientName = "Adam",
-                    ClientAddress = "Gardens",
-                    ContactName = "Bob",
-                    ContactEmail = "bob@gmail.com",
-                };
+                    var client = new Client     // Acts as an intialiser
+                    {
+                        ClientName = txt_clientName.Text,
+                        ClientAddress = txt_clientAddress.Text,
+                        ContactName = txt_contactName.Text,
+                        ContactEmail = txt_contactEmail.Text,
+                    };
 
-                context.Clients.Add(client);    // Add the client details
-                context.SaveChanges();  // Save it
+                    context.Clients.Add(client);    // Add the client details
 
-                MessageBox.Show("Saved!");
+                    context.SaveChanges();  // Save it
+
+                    MessageBox.Show("Saved!");
+                }
             }
         }
 
         private void btn_Exit_Click(object sender, EventArgs e)
         {
             this.Close();   // Closes the current form 'ClientDetailsEntryScreen'
         }
+
+        private void btn_View_Click(object sender, EventArgs e)
+        {
+            using (var context = new InvoiceDBContext())    // Using statement is discarded once the squishy brackets have been passed, saves memory
+            {
+                context.Database.EnsureCreated();   // Ensures database is created
+
+                var clients = context.Clients.ToList(); // Assign a bunch of client objects to a list
+
+                var viewModels = new List<ClientViewModel>();
+                foreach (var db in clients)
+                {
+                    viewModels.Add(ClientViewModel.FromDbModel(db));    // Add the client details to a list
+                }
+                dtaGridDetails.DataSource = viewModels; // Show all the client details on a datagrid
+                //dtaGridDetails.DataSource = context.Clients.ToList();
```

Comment:
Not a problem here, but it'll be worth noting that you should delete commented out code before committing to source control. You can take care of this in the next PR if you like.